### PR TITLE
fix: connection list

### LINKED ISSUE #656: the connection page does not display connections

### PR DESCRIPTION
**Pull request by notbot00**
# Related issues 🔗

Closes https://github.com/vegaprotocol/vegawallet-desktop/issues/656

# Description ℹ️

Making sure connections get displayed even when they have no assigned permissions to read any keys, etc.

### CODE REVIEW

File: libs/wallet-ui/src/components/connection-list/connection-list.tsx [hunk 1]
```diff
@@ -18,10 +18,12 @@ export const ConnectionList = ({ wallet }: ConnectionListProps) => {
   const connectionList = useMemo(() => {
     return Object.keys(wallet.connections || {}).filter((key) => {
       const c = wallet.connections?.[key]
-      return c && !(!c.active && c.permissions.publicKeys.access === 'none')
+      return c && !(!c.active && c.permissions.publicKeys?.access === 'none')
     })
   }, [wallet.connections])
 
+  console.log(wallet)
```
Comment (mattrussell36) [inline libs/wallet-ui/src/components/connection-list/connection-list.tsx]: chore (blocking): stray log

File: libs/wallet-ui/src/components/connection-list/connection-manage.tsx [hunk 2]
```diff
@@ -157,6 +157,21 @@ export const ManagePermissions = ({
     [client, dispatch, wallet.name, hostname, onClose]
   )
 
+  if (permissionAccessKeys.length === 0) {
+    return (
+      <div className="p-[20px]">
```
Comment (mattrussell36) [inline libs/wallet-ui/src/components/connection-list/connection-manage.tsx]: chore (blocking): Can we just use p-5, mb-5 etc, they are the same, lets just use the default tw config spacing wherever possible

File: libs/wallet-ui/src/components/connection-list/connection-manage.tsx [hunk 3]
```diff
@@ -157,6 +157,21 @@ export const ManagePermissions = ({
...
+        <p className="mb-[20px]">
+          <code>{hostname}</code> has not yet requested for any permissions.
+        </p>
+        <ButtonGroup className="pb-[20px]">
+          <ButtonUnstyled className="p-0" onClick={onClose}>
```
Comment (mattrussell36) [inline libs/wallet-ui/src/components/connection-list/connection-manage.tsx]: suggestion (non blocking): Now that we are using tw I think using a bare `<button>` element is better than the `<ButtonUnstyled>` as tw will unset all styles. Not sure if you need additional spacing/padding here though

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/wallet-ui/src/components/connection-list/connection-list.tsx (hunk 1 above)
Comment (notbot00) [reply libs/wallet-ui/src/components/connection-list/connection-list.tsx]:
```suggestion
```

File: libs/wallet-ui/src/components/connection-list/connection-manage.tsx (hunk 2 above)
Comment (notbot00) [reply libs/wallet-ui/src/components/connection-list/connection-manage.tsx]:
```suggestion
      <div className="p-5">
```

File: libs/wallet-ui/src/components/connection-list/connection-manage.tsx (hunk 3 above)
Comment (notbot00) [reply libs/wallet-ui/src/components/connection-list/connection-manage.tsx]: changed this, will most likely need to do some conflict resolution with your PR, as there are other unstyled button components around in this component

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**